profile/prune.go

```diff
@@ -130,7 +133,10 @@ func (p *Profile) PruneFrom(dropRx *regexp.Regexp) {
 				funcName := strings.TrimPrefix(fn.Name, ".")
 				// Account for unsimplified names -- trim starting from the first '('.
 				if index := strings.Index(funcName, "("); index > 0 {
-					funcName = funcName[:index]
+					// ... unless the thing in brackets is an anonymous namespace marker
+					if !strings.HasPrefix(funcName[index:], "(anonymous namespace)") {
```

Comment:
This should be a little more general -- just search for the first '(' that's not part of one of these reserved cases (operator() is another one). It will still fail for '(' inside a template, but I'm not sure there's an easy way around that other than actually parsing the AST.